src/core/ext/filters/client_channel/resolver/dns/c_ares/grpc_ares_wrapper.cc

```diff
@@ -145,6 +143,8 @@ void grpc_ares_complete_request_locked(grpc_ares_request* r) {
   ServerAddressList* addresses = r->addresses_out->get();
   if (addresses != nullptr) {
     grpc_cares_wrapper_address_sorting_sort(addresses);
+    GRPC_ERROR_UNREF(r->error);
```

Comment:
Hmmm... this means that if we do both an A and an AAAA lookup and one succeeds and the other fails, we ignore the failure.  I guess that makes sense, since if we have *any* backend addresses, it's better to return what we have than to ignore all of it.  But we should probably at least log the error somewhere.

Reply:
This is an interesting case, but in the example case that an A lookup fails, but the corresponding AAAA lookup succeeds, I think we want to only log the A lookup failure in the tracer.

The reason being is that it's legitimate to have an ipv4-only or ipv6-only service, and for e.g. an ipv6-only service, A record queries will always result in a no-error/no-data responses. Specifically, A record queries will come back with a DNS rcodes of `NOERROR`,  but they will have a zero-length ANSWER section. When c-ares parses such a response, it will return a [c-ares status of ENODATA](https://github.com/c-ares/c-ares/blob/master/ares_query.c#L165), and then we'll trace log a "DNS server returned answer with no data" log message. So, if we turn that trace log into e.g. an ERROR log, then users of ipv4-only and ipv6-only services will get spam logs.

Reply:
Makes sense.  I guess as long as we've got a tracer message for it, it's fine.